TlmCmd/block_command_loader.c

```diff
@@ -193,20 +194,26 @@ void BCL_tool_prepare_param_float(float val)
   block_command_loader_.params[block_command_loader_.param_idx++] = (uint8_t)((tmp      ) & 0xff);
 }
 
-// 実装時点ではuint64_tを使えなかったかつ使う人も少ないのでいったん無効化
-#if 0
-void BCL_tool_prepare_param_double(double val)
+void BCL_tool_prepare_param_uint64(uint64_t val)
 {
-  int i;
+  uint64_t tmp = val;
+  endian_memcpy(&block_command_loader_.params[block_command_loader_.param_idx], &tmp, sizeof(tmp));
+  block_command_loader_.param_idx += 8;
+}
 
-  // TODO: idx超過のassertionを入れる
-  for (i = 0; i < sizeof(val); i++)
-  {
-    block_command_loader_.param_buffer[block_command_loader_.command_idx][block_command_loader_.param_idx++]
-        = (unsigned char)((val >> (sizeof(val) - i) * 8) & 0xff);
-  }
+void BCL_tool_prepare_param_int64(int64_t val)
+{
+  int64_t tmp = val;
+  endian_memcpy(&block_command_loader_.params[block_command_loader_.param_idx], &tmp, sizeof(tmp));
+  block_command_loader_.param_idx += 8;
+}
+
+void BCL_tool_prepare_param_double(double val)
+{
+  double tmp = val;
+  endian_memcpy(&block_command_loader_.params[block_command_loader_.param_idx], &tmp, sizeof(tmp));
+  block_command_loader_.param_idx += 8;
```

Comment:
[IMO]

```suggestion
  block_command_loader_.param_idx += sizeof(tmp);
```

というのと，これってなんで tmp に一回保存してるんだっけ？

Reply:
値渡しなのでアドレスが存在しない可能性があるかなと思ったのですが、どうなんでしょうか？

Reply:
なるほどです、消しておきます